[tayian] iP

tayian opened the pull request:
GlenProMax
> “Think less, do more” – _original quote_


GlenProMax frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/tayian/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tayian committed:
Update and rename Duke.java to Glen.java

tayian committed:
Implement feature to greet user then exit

tayian committed:
Implement feature to echo commands entered by user, and exit when user types bye.

tayian committed:
Add feature to store text from user and display list when requested

tayian committed:
Add Task class

tayian committed:
Implement feature to mark and unmark task as done

tayian committed:
Add Deadline, Event, Todo (types of Tasks)

tayian committed:
Update Task to support Subtypes

tayian committed:
Support tracking different tasks  - Deadline, Event, Todo

tayian committed:
Add error handling to handle invalid inputs from user

tayian committed:
Implement feature to delete task from list

tayian committed:
Update EXPECTED.TXT

tayian committed:
Update input.txt

tayian committed:
Update runtest.sh

tayian committed:
Add automated text ui testing

tayian committed:
Add Storage class

tayian committed:
Update Deadline.java to support storage

tayian committed:
Update Event.java to support storage

tayian committed:
Update Todo.java to support storage

tayian committed:
Update Task.java to support subclasses

tayian committed:
Add storage feature to Glen

tayian committed:
Merge branch 'branch-Level-7'

tayian committed:
Add StringToDateTime.java to support Date and Time

tayian committed:
Merge branch 'branch-Level-8'

Add StringToDateTime.java to support Date and Time

longnguyentan commented on src/main/java/Glen.java:
```diff
@@ -23,7 +22,8 @@ public static void main(String[] args) {
                 try {
                     end = input.substring(++index);
-                } finally {
+                } catch (StringIndexOutOfBoundsException e) {}
```
I think it's small detail but `catch (StringIndexOutOfBoundsException e) {}` should be replaced with
```Java
try {
    # Your remaining code here
} catch (StringIndexOutOfBoundsException e) {
} finally {
    # Your remaining code here
}

longnguyentan reviewed (commented):
All look great 👍

kwangthiag commented on src/main/java/Glen.java:
```diff
@@ -0,0 +1,177 @@
+        }
+    }
+
+    static String lst() {
```
This could be changed to displayList or printList.

kwangthiag commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,62 @@
+        return (isDone ? "[X] " : "[ ] "); // mark done task with X
+    }
+
+    public String toString() {
```
Can add @Override.

kwangthiag reviewed (commented):
The names look mostly correct, keep up the good work!

leezhanpeng commented on src/main/java/Glen.java:
```diff
@@ -0,0 +1,177 @@
+
+                            System.out.println(addEvent(string1, string2, string3));
+                        } catch (Exception StringIndexOutOfBoundsException) {
+                            System.out.println(HORLINE + "\u2639 OOPS!!! Your request needs to be formatted as:\nevent <event name> /from <start time> /to <end time>\n" + HORLINE);
```
With long strings, maybe it will be good if you add linebreaks for better readability?

leezhanpeng reviewed (commented):
Overall nice work with your coding standards! Personally feel that some changes to improve readability will be nice, other than that, good job 👍🏻.

tayian committed:
Add Ui.java and Parser.java via upload

tayian committed:
Update Storage.java to support TaskList

tayian committed:
Add TaskList class

tayian committed:
Update Glen.java for more OOP

tayian committed:
Merge pull request from add-gradle-support

Add gradle support

tayian committed:
Add JavaDoc for TaskList.java

tayian committed:
Add JavaDoc for Task.java

tayian committed:
Add JavaDoc for Deadline.java

tayian committed:
Add JavaDoc for Event.java

tayian committed:
Add JavaDoc for Todo.java

tayian committed:
Add JavaDoc for Glen.java

tayian committed:
Merge pull request from branch-A-JavaDoc

Explain Classes and Methods with JavaDoc

tayian committed:
Update Ui.java to support Intro and Exit

tayian committed:
Merge pull request from branch-A-CodingStandard

Confirm that Project Follows Coding Standard

tayian committed:
Update TaskList.java to support Find

tayian committed:
Update Glen.java to support Find

tayian committed:
Merge branch 'branch-Level-9'

Implement Find feature

tayian committed:
Update Storage.java

updateTask method to support deletion of task as well (DRY principle)

tayian committed:
Refactor Glen.java for Better Code Organization

Refactoring Glen.java for Better Code Organization

The original Glen class was handling multiple responsibilities, including user interaction, parsing, task management, and storage. This made the code less modular and harder to maintain.

To improve the code organization, the class has been refactored into three separate classes: Glen, Ui, and Parser.

- Glen class now focuses on managing the flow of the chatbot, interacting with the user through the console, and orchestrating the operations between the Ui, Parser, TaskList, and Storage classes.
- Ui class is responsible for handling the user interface, including displaying introduction and exit messages.
- Parser class now handles the parsing of user input and delegates the corresponding tasks to the TaskList and Storage classes.

This separation of concerns leads to cleaner, more modular code, making it easier to understand, extend, and maintain in the future.

tayian committed:
Update TaskList.java to use instanceof

tayian committed:
Refactor type checks to utilize instanceof

Refactoring type checks to utilize instanceof -- improve code readability and maintainability

The original code used nested try-catch blocks with ClassCastException to determine the type of the 'task' object. This approach can be confusing and less efficient.

In the edited code, I replaced the try-catch blocks with a series of if-else statements that utilize the instanceof operator to check the type of 'task'. This change improves code readability and makes it easier to understand the logic. Additionally, it provides better performance as exceptions are costly in terms of computational resources.

This commit improves code quality by enhancing readability, maintainability, and efficiency. It adheres to best practices for type checking and helps prevent potential exceptions, resulting in more robust code.

tayian committed:
Add Java assertions to affirm correctness of assumptions

The code lacks Java assertions, making it challenging to document assumptions which are critical at various points in the code.

Adding Java assertions is crucial for documenting assumptions. It
helps identify and also prevent unexpected issues during
development and runtime.

This commit introduces Java assertions at key points in the codebase.

This enhances code clarity and maintainability, and improves code readability across multiple classes.

tayian committed:
Add Java assertions to affirm correctness of assumptions

The code lacks Java assertions, making it challenging to document assumptions which are critical at various points in the code.

Adding Java assertions is crucial for documenting assumptions. It helps identify and also prevent unexpected issues during development and runtime.

This commit introduces Java assertions at key points in the codebase.

This enhances code clarity and maintainability, and improves code readability across multiple classes.

tayian committed:
Merge branch 'branch-A-Assertions'

The code lacks Java assertions, making it challenging to document assumptions which are critical at various points in the code.

Adding Java assertions is crucial for documenting assumptions. It helps identify and also prevent unexpected issues during development and runtime.

This commit introduces Java assertions at key points in the codebase.

This enhances code clarity and maintainability, and improves code readability across multiple classes.

tayian committed:
Update and rename src/main/java/Deadline.java to src/main/java/glen/task/Deadline.java

tayian committed:
Update and rename src/main/java/Event.java to src/main/java/glen/task/Event.java

tayian committed:
Update and rename src/main/java/Task.java to src/main/java/glen/task/Task.java

tayian committed:
Update and rename src/main/java/Todo.java to src/main/java/glen/task/Todo.java

tayian committed:
Update and rename src/main/java/Parser.java to src/main/java/glen/util/Parser.java

tayian committed:
Update and rename src/main/java/Storage.java to src/main/java/glen/util/Storage.java

tayian committed:
Update and rename src/main/java/StringToDateTime.java to src/main/java/glen/util/StringToDateTime.java

tayian committed:
Update and rename src/main/java/TaskList.java to src/main/java/glen/util/TaskList.java

tayian committed:
Update and rename src/main/java/Ui.java to src/main/java/glen/util/Ui.java

tayian committed:
Update and rename src/main/java/Glen.java to src/main/java/glen/Glen.java

tayian committed:
Organise the classes into suitable Java packages

Refactor project directory structure for better organization

The current project structure lacks a clear organization, making it difficult to navigate and maintain. To address this issue, the directory structure has been reorganized.

Moved all .java files into a 'duke' package to encapsulate the main application logic and make it more modular.
Created 'task' sub-package to house Deadline, Event, Task, Todo
Also created 'util' sub-package to house Parser, Storage, StringToDateTime, TaskList, Ui, to provide a more structured approach to managing utility classes.
This reorganization follows Java's recommended package naming conventions, improving code readability and maintainability.

Additionally, this change enables better scalability for future feature additions or modifications.

tayian committed:
Add JUnit tests for Parser, Storage, and TaskList

The current codebase lacks automated tests, which makes it prone to regressions and hard to maintain.

By adding JUnit test classes for the mentioned classes, we can ensure that the code functions as expected and catch any potential issues early in development.

I have created separate JUnit test classes for each of the classes to thoroughly test their functionality. Each test class contains test cases for various scenarios, including edge cases and boundary conditions. This approach ensures comprehensive coverage of the codebase.

Additionally, I have followed best practices for writing unit tests, including using descriptive test method names, organizing tests into logical groups, and using assertions to validate expected outcomes. This makes the tests easy to read, understand, and maintain. This commit also includes the necessary setup for running the tests, such as adding dependencies to the build file and configuring the test runner.

Overall, this approach will lead to more robust and reliable code, making it easier to develop and maintain the application in the long run.

tayian committed:
Add GUI to Glen application

Add GUI for enhanced user interaction and visual representation

This commit introduces a graphical user interface (GUI) to replace the existing text-based interface, providing users with a more intuitive and visually appealing interaction with the application. The GUI integrates seamlessly with the underlying logic, maintaining all existing functionality while enhancing the user experience.

Changes Made:

- Implemented GUI components using JavaFX
- Ensured compatibility with existing backend logic

tayian committed:
Add User Guide and GUI Screenshots

Included a comprehensive user guide to provide clear instructions on using the application effectively. Added screenshots of the Graphical User Interface (GUI) for visual reference.

Details:
- Utilized 'docs' directory to store the user guide.
- Authored the user guide in Markdown format, covering usage instructions.
- Captured screenshots of critical application screens and saved them in 'docs'.
- Ensured that the guide is accessible and user-friendly.

tayian committed:
Update build.gradle to support Java 11

tayian committed:
Branch level 10

Properly merge branch level 10
